clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -522,7 +522,16 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         long updatedNow = this.time.milliseconds();
         List<ClientResponse> responses = new ArrayList<>();
         handleCompletedSends(responses, updatedNow);
-        handleCompletedReceives(responses, updatedNow);
+
+        try {
+            handleCompletedReceives(responses, updatedNow);
+        } catch (StaleMetadataException e) {
```

Comment:
is "StaleMetadataException" thrown only in validateCluster? is there any other case that could end up with StaleMetadataException?  If so, maybe we should have a new type of exception. Can you add a unit test for it?